src/Particle/Par_Output_TracerParticle_Mesh.cpp

```diff
@@ -116,7 +120,7 @@ void Par_Output_TracerParticle_Mesh()
             if ( TVar & AllVar_PreparePatch )
                Prepare_PatchData( lv, TimeNew, Var, NULL, ParGhost, 1, &PID0, TVar, _NONE,
                                   OPT__FLU_INT_SCHEME, INT_NONE, UNIT_PATCH, NSIDE_26, IntPhase_No,
-                                  OPT__BC_FLU, BC_POT_NONE, MinDens_No, MinPres_No, MinTemp_No, MinEntr_No, DE_Consistency_No );
+                                  OPT__BC_FLU, OPT__BC_POT, MinDens_No, MinPres_No, MinTemp_No, MinEntr_No, DE_Consistency_No );
```

Comment:
[Bug] `OPT__BC_POT` is only defined when enabling `GRAVITY`.

Reply:
This bug is fixed in the latest commit. 

However, when `GRAVITY` is disabled (and so for `MASSIVE_PARTICLES`), the linker will fail due to the functions `Prepare_PatchData_FreeParticleDensityArray()` and `Prepare_PatchData_InitParticleDensityArray()` are not defined, which are called in `Aux_ComputeProfile.cpp`, `Aux_FindExtrema.cpp`, and `Aux_FindWeightedAverageCenter.cpp`. Should I also fix it in this PR?

Reply:
@hfhsieh Please file a separate PR to fix this issue. I think we should use `MASSIVE_PARTICLES` instead of `PARTICLE` in those files.